pkg/ipamd/ipamd.go

```diff
@@ -647,16 +647,18 @@ func (c *IPAMContext) increaseDatastorePool() {
 	ipamdActionsInprogress.WithLabelValues("increaseDatastorePool").Add(float64(1))
 	defer ipamdActionsInprogress.WithLabelValues("increaseDatastorePool").Sub(float64(1))
 
-	short, _, warmTargetDefined := c.datastoreTargetState()
-	if warmTargetDefined && short == 0 {
+	short, _, warmIPTargetDefined := c.datastoreTargetState()
+	if warmIPTargetDefined && short == 0 {
 		log.Debugf("Skipping increase Datastore pool, warm target reached")
 		return
 	}
 
-	short, warmTargetDefined = c.datastorePrefixTargetState()
-	if warmTargetDefined && short == 0 {
-		log.Debugf("Skipping increase Datastore pool, warm prefix target reached")
-		return
+	if !warmIPTargetDefined {
+		shortPrefix, warmTargetDefined := c.datastorePrefixTargetState()
+        	if warmTargetDefined && shortPrefix == 0 {
```

Comment:
How about when `WARM_PREFIX_TARGET` is not defined (i.e.,) when `warmTargetDefined` is `false`? Shouldn't we return in that scenario as well? It appears that we're going to continue further if that is the case with the current logic.

Reply:
This is added to give WARM_IP_TARGET precedence over WARM_PREFIX_TARGET. If WARM_PREFIX_TARGET is not defined then tryAssignCidr will allocate just one prefix.